### DISCUSSION TITLE
How to: Your station’s media folder in the file manager? Easy.

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6510] [Show and tell]
## Ever wanted simple access to your station’s media folder in your file manager?

Linux provides. Simply use `sshfs` and you can add, delete, update your media as if it were local files in your file manager! (A little slower, though, of course, since files are actually remote.)

### What you need

- An SFTP user set up in AzuraCast that can access the station’s media.
- A Linux client system (your usual Linux desktop, laptop, or tablet). I will describe the setup for Debian-based systems here (Debian, Ubuntu, Linux Mint).

### What we will set up

- A local folder inside your user’s "Music" folder, called the same as your station name.
- Install `sshfs` and show how to use it manually.
- Mount your station media into said folder, so you have secure access to all files.
- Use `ssh` options to avoid inactivity disconnections.
- Unmount the folder after work is done (=terminate the connection).

### What I won’t show

- Automating file transfers.
- Show how to automount the folder on boot (in fstab). This might pose a security risk.

### Install `sshfs`

`sshfs` is a **s**ecure **sh**ell **f**ile **s**ystem that works over SFTP and provides mounting remote files and folders into your file system, as if they were local. Since SFTP is the underlying transport, all transfers through the Internet are safe and encrypted.

Open a terminal and install:
```bash
sudo apt install sshfs
```

### Prepare a folder to use for your remote media

We will use a subfolder of our local `~/Music` folder here, named after the station:
```bash
mkdir "~/Music/Nite Radio"
```

Use your station name instead of `Nite Radio`.

Note this folder will appear empty if your station isn’t mounted "over" it.

You can of course also use a folder under the more traditional mount point `/mnt`. It just eases the workflow (and makes it a little more secure) to mount your station media under `~/Music/Station Name`.

### Mount the remote station media into our new folder

_(Edited for safety: Use the logged-in user, not root.)_

- Use your AzuraCast’s host name instead of `example.com`, and its configured SFTP port (usually `2022`).
- Use the SFTP user name and credentials you created in AzuraCast instead of `testuser`.
- Use your "station name" folder name instead of `Nite Radio`.

```bash
sshfs -p 2022 testuser@example.com:/ "~/Music/Nite Radio"
testuser@example.com's password: 
```
Type testuser’s password and press `Enter`. (You won’t see anything, or just *.)

On the first connection, you will get asked to verify the server’s identity and continue:
```
The authenticity of host '[example.com]:2022 ([179.37.29.88]:2022)' can't be established.
ED25519 key fingerprint is SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
```
Type `yes` and press `Enter`.

You should now see your station’s media folder in `~/Music/Nite Radio` and be able to add, delete and change files!

![Arbeitsfläche 1_022](https://github.com/AzuraCast/AzuraCast/assets/3706922/15b485c5-9e8a-49b0-922f-b01bf5c96dbf)

**Note:** Depending on the settings of your system, the connection might be terminated after a while of inactivity. This is for security reasons—just mount again.

### Avoiding disconnections due to inactivity

Sometimes you plan to work on your station’s media folder for quite a while, and the inactivity disconnections break your workflow.

No problem: We can use `ssh` options (which would otherwise be in `/etc/ssh/ssh_config`) to send a periodic "keep alive" message, thus avoiding automatic disconnects. Using the command from above, let’s add a "keep alive" every 15 s, and the folder being unmounted only after _two_ such keep-alive messages have failed:

```bash
sshfs -p 2022 testuser@example.com:/ "~/Music/Nite Radio" -o ServerAliveInterval=15 -o ServerAliveCountMax=2
```

You will observe the folder now staying mounted until you manually unmount it (or a network problem occurs).

### Unmounting/terminating the connection

_(Edited: Use logged-in user instead of root.)_

Simply use the `umount`or `fusermount -u` command to unmount your folder safely and terminate the connection to your station:

```bash
umount "/home/matthias/Musik/Nite Radio"
```
or
```
fusermount -u "/home/matthias/Musik/Nite Radio"
```

Note the use of `"` in case the folder path contains spaces.

Also note that umount _won’t_ honor the tilde (`~`=home folder) expansion, so you _must_ use "/home/matthias/Musik/Nite Radio" here (substitute _your_ username for "matthias"). Or use the shell’s _tab completion_ which would make it something like `~/Musik/Nite\ Radio`—the space getting escaped.

Having your station media folder mounted as a regular, logged-in user has the additional advantage that you can unmount it by simply clicking the ⏏ (eject/unmount) button in your file manager:

![Nite Radio_001](https://github.com/AzuraCast/AzuraCast/assets/3706922/246dd9d6-824f-4a1c-91b9-f463831d8250)

---

**Enjoy** easy station media access!

If you find this useful, _upvote_ by using the blueish up-arrow button below this post, _not_ by using the :+1: emoji. (You can still do that, but it won’t upvote the discussion, just give me a warm feeling…)

**Comment by BusterNeece** [comment]
@Moonbase59 I added a link to this thread on the SFTP Server docs page, along with links to some proprietary software that make this process easy on Windows/MacOS: https://docs.azuracast.com/en/user-guide/sftp-server#mounting-as-a-directory-on-your-computer

**Reply by Moonbase59** [reply]
A good thing! I tend to write down such things, because non-technical users often don’t structure their workflows well, or are not aware of how easy things can often be.

**Reply by mdrxy** [reply]
@BusterNeece You may want to consider changing the wording on that page from "[SSHFS for Linux](https://github.com/libfuse/sshfs)" to "[SSHFS for Linux/macOS](https://github.com/libfuse/sshfs)" since it works on both!

**Reply by BusterNeece** [reply]
@mdrxy Updated!

**Reply by Moonbase59** [reply]
Good to know it works on MacOS, too! Thanks for the info!

**Comment by Moonbase59** [comment]
In my original post I used `sudo sshfs` which is only necessary in special cases. Using `sshfs` as a regular user is safer, so I’m going to update the first post.

**Comment by Moonbase59** [comment]
Added a section on how to avoid inactivity disconnections.
